proxmoxtf/resource/firewall/rules.go

```diff
@@ -222,81 +217,32 @@ func RulesCreate(ctx context.Context, api firewall.Rule, d *schema.ResourceData)
 		return diags
 	}
 
-	// reset rules, we re-read them (with proper positions) from the API
-	err := d.Set(MkRule, nil)
-	if err != nil {
-		return diag.FromErr(err)
-	}
-
 	d.SetId(api.GetRulesID())
 
 	return RulesRead(ctx, api, d)
 }
 
-// RulesRead reads rules from the API and updates the state.
+// RulesRead gets updates the tf state.
+//
+//nolint:revive,unused-parameter //bypass unused-parameter warning as ctx is needed for invokeRuleAPI.
 func RulesRead(ctx context.Context, api firewall.Rule, d *schema.ResourceData) diag.Diagnostics {
 	var diags diag.Diagnostics
 
-	readRule := func(pos int, ruleMap map[string]interface{}) error {
```

Comment:
Hm.. I didn't quite get this change. The `RulesRead` supposed to read rules from PVE API. Now it reads back only from the state 🤔

Reply:
Hi, sure let me explain:

```rulesCreate``` function was unnecessarily resetting the rule state before calling ```RulesRead```. This was redundant because the rules were already created in specific order in one take within ```rulesCreate``` and represented correctly in ```d```. However, the correct ```pos``` value for each rule was not being set at that time.

By removing the state reset at the end of ```rulesCreate```, rules are represented in the correct (in terms of user configuration) order. 
Calling ```RulesRead``` we can iterate over the rules and for each rule id increment the POS value for each rule to set the actual rule POS. 
This ensures that the rules, and their position are in the correct order, even without reading them from the API.

Reply:
Although this proposal cannot handle changes outside of Terraform, does stabilizes the terraform plan / apply / delete.
Next step though is the rule comparison in terms of API/state-configuration ,so outside changes to be reverted and to be ensured the terraform configuration.